pymatgen/io/lammps/inputs.py

```diff
@@ -222,6 +253,61 @@ def add_stage(self, command: str | list | dict, stage_name: str | None = None):
         else:
             raise TypeError("The command should be a string, list of strings or dictionary.")
 
+    def remove_stage(self, stage_name: str):
+        """
+        Removes a whole stage from the LammpsInputFile.
+
+        Args:
+            stage_name (str): name of the stage to remove.
+        """
+        if stage_name in self.stages_names:
+            self.list_of_commands.pop(self.stages_names.index(stage_name))
+            self.nstages = self.get_nstages()
+            self.ncomments = self.get_ncomments()
+        else:
+            raise LookupError("The given stage name is not present in this LammpsInputFile.")
+
+    def merge_stages(self, stage_names: list):
+        """
+        Merge multiple stages of a LammpsInputFile together.
+        The merged stage will be at the same index as the first of the stages to be merged.
+        The others will appear in the same order as provided in the list. Other non-merged stages will follow.
+
+        Args:
+             stage_names (list): list of strings giving the names of the stages to be merged.
+        """
+        if not all([stage in self.stages_names for stage in stage_names]):
+            raise ValueError("At least one of the stages to be merged is not in the LammpsInputFile.")
+
+        indices_stages_to_merge = [self.stages_names.index(stage) for stage in stage_names]
+        if not np.all([np.array(indices_stages_to_merge[1:]) >= np.array(indices_stages_to_merge[:-1])]):
+            raise ValueError(
+                """The provided stages are not in the right order. You should merge stages
+            order of appearance in your LammpsInputFile. If you want to reorder stages, use add_stage and
+            remove_stage."""
+            )
+
+        list_of_commands = self.list_of_commands[: indices_stages_to_merge[0]]
+        stages_names = self.stages_names[: indices_stages_to_merge[0]]
+
+        merge_name = "Merge of: " + ", ".join([self.stages_names[i] for i in indices_stages_to_merge])
+        merged_commands = []
+        for i in indices_stages_to_merge:
+            for j in range(len(self.list_of_commands[i][1])):
+                merged_commands.append(self.list_of_commands[i][1][j])
+
+        merged_stages = [merge_name, merged_commands]
+        stages_names.append(merge_name)
+        list_of_commands.append(merged_stages)
+
+        for i_stage, stage in enumerate(self.list_of_commands):
+            if i_stage > indices_stages_to_merge[0] and i_stage not in indices_stages_to_merge:
+                list_of_commands.append(stage)
+
+        self.list_of_commands = list_of_commands
```

Comment:
What do you think would be more useful: that the method returns the stage(s) that has/have been removed, the entire initial LammpsInputFile, or something else ? Or that they are stored as an attribute of the object, like LammpsInputFile.backup_list_of_commands ? Just trying to find what would be the best solution for the user.

Reply:
Wouldn't they just start from scratch in such a case? I am currently thinking about a possible scenario where one would like to continue after an error in a script. Sorry, if I did not get it correctly.

Reply:
@JaGeo not continue after the error, but go back one step before the error occurred and re-start (with different settings) from there. Then, I think it's easier to have a direct access to what has been changed instead of having to look at the files before and after removing/merging.
@gbrunin I think it's the easiest to return the stages that were removed and store the information somehow. 
But I'm not a LAMMPS user at all, if you see no necessity for any kind of stage backup, it's completely fine.

Reply:
Thinking more about it, I think this is something that should be left to the user or to third-party codes that will use it (e.g., atomate2). In case of error, you can always restart the process of creating the LammpsInputFile. In other cases (e.g., workflow creation), you can always save the LammpsInputFile in another variable to use it again later. 
If we decide to save previous states of the LammpsInputFile, it might become uneasy to handle and not very readable.

Reply:
I agree. This should not be part of the io module.